BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java

```diff
@@ -0,0 +1,284 @@
+	private WebElement buyNowBtn;
+
+	@FindBy(css = ".ps-wtb")
+	private WebElement buyNowModal;
+
+	@FindBy(xpath = "//h2[normalize-space()='Benadryl Allergy Ultratabs Tablets, 48 Count']")
+	private WebElement buyNowModalHeading;
+
+	@FindBy(css = ".jump-to-title")
+	private WebElement jumpToTitle;
+
+	@FindBy(css = "[href=\"#product-overview\"] > h2")
+	private WebElement prodOverviewHeading;
+
+	@FindBy(css = ".pane-node-field-product-directions")
+	private WebElement directionSection;
+
+	@FindBy(css = ".pane-node-field-product-directions h2")
+	private WebElement directionHeading;
+
+	@FindBy(id = "mini-panel-product_ingredients_modal")
+	private WebElement ingredientsSection;
+
+	@FindBy(css = "#mini-panel-product_ingredients_modal h2")
+	private WebElement ingredientsHeading;
+
+	@FindBy(css = "li #used-for")
+	private WebElement usedForSection;
+
+	@FindBy(id = "tab_used-for")
+	private WebElement usedForHeading;
+
+	@FindBy(css = "li #warnings")
+	private WebElement warningsSection;
+
+	@FindBy(id = "tab_warnings")
+	private WebElement warningsHeading;
+
+	@FindBy(id = "mini-panel-product_faq")
+	private WebElement faqsSection;
+
+	@FindBy(css = "#mini-panel-product_faq span")
+	private WebElement faqsHeading;
+
+	@FindBy(css = "#bv-product-reviews")
+	private WebElement reviewsSection;
+	
+	/**
+	 * Returns the WebElement of a 'Jump To' heading based on its text.
+	 *
+	 * @param headingText The text of the 'Jump To' heading.
+	 * @return WebElement The WebElement of the specified 'Jump To' heading.
+	 */
+	public WebElement getProductName(String path) { 
+		 return	 driver.findElement(By.xpath("//span[@class='node__title']//a[@href='/products/" + path + "']"));
+	}	
+	
+	/**
+	 * Returns the WebElement of a 'Jump To' heading based on its text.
+	 *
+	 * @param headingText The text of the 'Jump To' heading.
+	 * @return WebElement The WebElement of the specified 'Jump To' heading.
+	 */
+	public WebElement getJumpToHeadings(String headingText) { 
+		 return	 driver.findElement(By.xpath("//div[@class='jump-to-wrapper']//ul//li//a[text()='" + headingText + "']"));
+	}
+	
+	/**
+	 * Returns the WebElement of the Write a Review modal.
+	 */
+	public WebElement getWriteAReviewHeading() { 
+		 return	 driver.findElement(By.id("shadow-root")).getShadowRoot().findElement(By.cssSelector("#bv-ips-loading-text_bv-ips-title > span.ips__sc-r8pfgm-3.fINXNs"));
+	}
+	
+	/**
+	 * Navigates to a product detail page from the homepage by clicking on the product header menu 
+	 * and a specific product title.
+	 * 
+	 * @param link of the product url
+	 */
+	public void visitDetailsPage(String path) {
+	    Action.performActionwithExtentInfoLog(prodHeaderMenu, "click", "Clicking on : " + prodHeaderMenu.getText());
+	    Action.performActionwithExtentInfoLog(getProductName(path), "click", "Clicking on : " + prodTileTitle.getText());
+	    Action.verifyPageUrl("products/" + path);
```

Comment:
@zahinSJI why you sending text "products/" seperately??? send this from data file only

Reply:
Changes made